zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/NotebookTest.java

```diff
@@ -1232,7 +1234,7 @@ public void testPerSessionInterpreterCloseOnNoteRemoval() throws IOException, In
       setting.getOption().setPerNote(setting.getOption().SCOPED);
       notebook.getInterpreterSettingManager().restart(setting.getId());
     }
-
+    Thread.sleep(2000);
```

Comment:
Why this change ?

Reply:
I add this lines during debugging because of failed tests. But this was before adding this [magic line](https://github.com/apache/zeppelin/pull/3925/files#diff-87d8954289365d4fc9ab53cd9b2257ac28cbdef284ec83565e4d5bf9bafc6eedR248), which removes the shutdownhook thread. I'll try to revert the sleep statements.

I added these lines during debugging because of failed tests. But that was before adding this [magic line](https://github.com/apache/zeppelin/pull/3925/files#diff-87d8954289365d4fc9ab53cd9b2257ac28cbdef284ec83565e4d5bf9bafc6eedR248) that removes the shutdownhook thread in case of a regular shutdown. I will try to remove the sleep statements.

Reply:
I have removed the Sleep Statements.